### PR TITLE
Dark Bremsstrahlung backport to 10_6_X

### PR DESCRIPTION
**Pull request by revering**
Backport of implementation of Dark Bremsstrahlung physics process from #38329 to CMSSW_10_6_X for run II MC generation. 

CustomPhysics.cc in 10_6_30_patch1 is different from original CMSSW_12_5_X base, but git cms-rebase-topic successfully reapplied the necessary additions for the Dark Brem sim without including other changes. Needed to resolve a merge conflict in CustomPhysics/BuildFile.xml caused by differences in the spacing between \<use name> and the ordering of includes between 10_6_X and 12_5_X. Used the 12_5_X spacing and order, should have no effect (as far as I'm aware).

Generated a small test sample with Dark Brem enabled, and the process biases the cross section and produces Dark photons as expected.

### CODE REVIEW

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
+1

**Comment by civanch** [comment]
@revering , please add in the beginning of the title of this PR: [10_6_X]

File: SimG4Core/CustomPhysics/src/G4muDarkBremsstrahlungModel.cc [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+  G4int nElements = material->GetNumberOfElements();
+  const G4ElementVector* theElementVector = material->GetElementVector();
+  const G4double* theAtomNumDensityVector = material->GetAtomicNumDensityVector();
+  G4DataVector* dv = new G4DataVector();
+  G4double cross = 0.0;
+
+  for (G4int i = 0; i < nElements; i++) {
+    cross += theAtomNumDensityVector[i] *
+             ComputeCrossSectionPerAtom(
+                 particle, kineticEnergy, (*theElementVector)[i]->GetZ(), (*theElementVector)[i]->GetA(), cut);
```
Comment (civanch) [inline SimG4Core/CustomPhysics/src/G4muDarkBremsstrahlungModel.cc:315]: @revering , better now than never. (*theElementVector)[i]->GetA() returns molar mass of the atom (see https://geant4.kek.jp/lxr/source/materials/include/G4Element.hh). You need to change to
(*theElementVector)[i]->GetN(). The same patch should be added in 12_4 and 12_5.

File: SimG4Core/CustomPhysics/src/G4muDarkBremsstrahlungModel.cc [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+void G4muDarkBremsstrahlungModel::SampleSecondaries(std::vector<G4DynamicParticle*>* vdp,
+                                                    const G4MaterialCutsCouple* couple,
+                                                    const G4DynamicParticle* dp,
+                                                    G4double tmin,
+                                                    G4double maxEnergy)
+//Simulates the emission of a dark photon + electron. Gets an energy fraction and Pt from madgraph files. Scales the energy so that the fraction of kinectic energy is constant, keeps the Pt constant. If the Pt is larger than the new energy, that event is skipped, and a new one is taken from the file.
+{
+  //Deactivate the process after one dark brem. Needs to be reactivated in the end of event action. If this is in the stepping action instead, more than one brem can occur within each step.
+  G4bool state = false;
+  G4String pname = "muDBrem";
```
Comment (civanch) [inline SimG4Core/CustomPhysics/src/G4muDarkBremsstrahlungModel.cc:330]: @revering , it is a minor thing but better use G4 mass (avoiding problems in corner cases):

double muMass = dp->GetDefinition()->GetPDGMass(); // this is exact muon mass in internal units (MeV)

**Comment by civanch** [comment]
@revering , I put two comments to the model. Better to implement this just now and forward port in new PRs to 12_4 and 12_5.

**Comment by qliphy** [comment]
backport of #38329

**Comment by qliphy** [comment]
@revering   Please note there is a comment raised in master PR: https://github.com/cms-sw/cmssw/pull/38329#issuecomment-1202841384

**Comment by qliphy** [comment]
#38949  needs to be backported here also.  @revering

**Comment by qliphy** [comment]
please test

**Comment by civanch** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**